### DISCUSSION TITLE
update on each keydown.enter

### DISCUSSION THREAD

**Discussion by Grazulex** [livewire/livewire#5224] [1. Help]
Hello,

I have made a small livewire compoent with a input field without form. I need this input only to read datamatrix.
Here is my code : 
`<input wire:model="datamatrix" type="text"  wire:keydown.enter="searchDatamatrix" id="datamatrix" class="block w-full transition duration-75 rounded-lg shadow-sm focus:border-primary-600 focus:ring-1 focus:ring-inset focus:ring-primary-600 disabled:opacity-70 dark:bg-gray-700 dark:text-white dark:focus:border-primary-600 border-gray-300 dark:border-gray-600">`

everything work fine but slow... when I check in the network debug tools, I see a call each time when on press on a key in this input. The fonction "searchDatamatrix" is running only when I press the enter key but when  I press a other key,  I see call to the component with this data : 

**call** : http://localhost/livewire/message/incoming-plate

**payload** : 

`s":null,"datamatrix":"gdfgdgdccccv","cod":"","type":"cod","cod_is_disable":true,"tableRecordsPerPage":null,"page":1,"paginators":{"page":1},"isTableReordering":false,"tableColumnSearchQueries":[],"tableSearchQuery":"","selectedTableRecords":[],"tableSortColumn":null,"tableSortDirection":null,"toggledTableColumns":[],"mountedTableAction":null,"mountedTableActionData":[],"mountedTableActionRecord":null,"mountedTableBulkAction":null,"mountedTableBulkActionData":[],"tableFilters":null,"componentFileAttachments":[],"mountedFormComponentAction":null,"mountedFormComponentActionData":[],"mountedFormComponentActionComponent":null},"dataMeta":{"models":{"incoming":{"class":"App\\Models\\Incoming","id":9,"relations":[],"connection":"mysql","collectionClass":null}}},"checksum":"f3045fa7ac8e9bae70c26076ca3fb3dca8f5f11a65bccd38a5e74dfeda8c9635"},"updates":[{"type":"syncInput","payload":{"id":"o589f","name":"datamatrix","value":"gdfgdgdccccvfbf"}}]}`

**response** : 

`{"effects":{"html":null,"dirty":[]},"serverMemo":{"data":{"plates":null,"datamatrix":"gdfgdgdccccvfbf","tableRecordsPerPage":null,"tableSortColumn":null,"tableSortDirection":null,"mountedTableAction":null,"mountedTableActionRecord":null,"mountedTableBulkAction":null,"tableFilters":null,"mountedFormComponentAction":null,"mountedFormComponentActionComponent":null},"checksum":"d075174072f4eecdb998bd56aeb40adf9f88c4f6df3d632049e9ec11915a47e1"}}`

I have the feeling that everyting is slow due all there calls....
How can I prefend this type of situation ?

Thanks you

**Comment by Martleby** [comment]
Hello!

You might want to try:

```blade
wire:model.defer="datamatrix"
```
https://laravel-livewire.com/docs/2.x/properties#deferred-updating

or

```blade
wire:model.debounce="datamatrix"
```
https://laravel-livewire.com/docs/2.x/properties#debouncing-input

Cheers,
Martin.
